vnext/ReactUWP/Views/ShadowNodeBase.cpp

```diff
@@ -56,13 +59,21 @@ void ShadowNodeBase::RemoveChildAt(int64_t indexToRemove)
 
 void ShadowNodeBase::onDropViewInstance()
 {
+  m_handledKeyboardEventHandler = nullptr;
 }
 
 void ShadowNodeBase::ReplaceView(XamlView view)
 {
+
   SetTag(view, GetTag(m_view));
 
   m_view = view;
+
+  if (m_handledKeyboardEventHandler)
+  {
+    m_handledKeyboardEventHandler->unhook();
+    m_handledKeyboardEventHandler->hook(view);
```

Comment:
This is a slightly dangerous pattern - it requires the caller to know what situations unhook must be called before calling hook().  Getting this pattern wrong could cause unpredictable behavior.  I suggest:
-Make unhook check for null and only call revoke if the revoker(s) are null.
-Call unhook from hook() at the beginning of the function, so that it's impossible to call hook in a dangerous way.

Reply:
KeyboardEventHandler is already smart enough and it will not cause any problem even if I didn't call unhook here.

unhook here is to clearly express that I unhook from the old view and hook to the new view.

Reply:
Sorry I missed this detail.

---
In reply to: [295598865](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295598865) [](ancestors = 295598865)

Reply:
If it's not necessary why do it? We already had 2 reviewers trip up on this and leave a comment.

Reply:
The comment is to make m_handledKeyboardEventHandler be smart. otherwise it may cause problem if other one is using it.
m_handledKeyboardEventHandler is already very smart and you don't need to unhook, then hook.
I don't see any benefit I have a comment for unhook here.
No change.

---
In reply to: [295934785](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295934785) [](ancestors = 295934785)